### PR TITLE
Allow to use an already open socket for the ssh connection. 

### PR DESCRIPTION
**Pull request by scathaig**
This is especially useful for NETCONF Call Home (https://tools.ietf.org/html/draft-ietf-netconf-reverse-ssh-05) connections where the server opens a tcp connection to the client and the client opens a NETCONF session using the already open tcp connection.

### CODE REVIEW

**Comment by ntwrkguru** [comment]
@leopoul Can you review this please?

File: ncclient/transport/ssh.py [hunk 1]
```diff
@@ -168,7 +168,12 @@ def connect(self, host, port=830, timeout=None, unknown_host_cb=default_unknown_
         *look_for_keys* enables looking in the usual locations for ssh keys (e.g. :file:`~/.ssh/id_*`)
 
         *ssh_config* enables parsing of an OpenSSH configuration file, if set to its path, e.g. ~/.ssh/config
+
+        *sock* is an already open socket which shall be used for this connection. Useful for NETCONF Call Home.
         """
+        if not (host or socket):
```
Comment (einarnn) [inline ncclient/transport/ssh.py]: Shouldn't `socket` here be `sock`?

**Comment by stacywsmith** [comment]
@mwiget Should this pull request be closed in favor of your #196 request?

**Comment by mwiget** [comment]
@stacywsmith imho yes. I cherry-picked this pull request as the base of the already merged #196 .

**Comment by stacywsmith** [comment]
@scathaig Can you please review the changes in #196. I believe they are superset of this PR. If so, could you please close this request?

**Comment by einarnn** [comment]
@scathaig - No response from you to queries from @stacywsmith, so closing this PR for now as changes cannot be reviewed and merged in current state. Please update PR if you feel there is still changes you would like to see!

Cheers,

Einar

**Pull request closed without merging**